lib/archethic/contracts/interpreter/scope.ex

```diff
@@ -8,6 +8,8 @@ defmodule Archethic.Contracts.Interpreter.Scope do
   """
   @spec init(map()) :: :ok
   def init(global_variables \\ %{}) do
```

Comment:
Scope.init is not used anymore outside of Scope, you can make it private

Reply:
i can't or it won't be tested

Reply:
You can test it in the test of execute_ast function

Reply:
actually the execute function wasn't tested but i don't know what tests I should add

Reply:
You can create some AST to be executed as a function, create a scope containing this function and some global variable, call Scope.execute_function and ensure the function is well executed and return expected result